discordbot/commands/drc.py

```diff
@@ -29,70 +30,83 @@ def format_deck(x: Dict) -> Dict:
     if count <= MAX_DECKS_SHOWN:
-        arr = [format_deck(x) for x in output['data']]
+        arr = [format_deck(x) for x in data['sample']]
     else:
-        data = output['data'][:MAX_DECKS_SHOWN_WITH_CONTINUATION]
-        arr = [format_deck(x) for x in data]
-        arr.append({'name': 'Other results', 'value': '[{n} more results found]({domain}/minimize/{url})'.format(
+        subsample = data['sample'][:MAX_DECKS_SHOWN_WITH_CONTINUATION]
+        arr = [format_deck(x) for x in subsample]
+        arr.append({'name': 'Other results', 'value': '[{n} more results found]({domain}{path}?q={query}{pd})'.format(
             domain=link_domain,
             n=count - MAX_DECKS_SHOWN_WITH_CONTINUATION,
-            url=output['compress'],
+            path='/deck-search',
+            pd='&dist=penny_dreadful',
+            query=query.replace(' ', '%20'),
         )})
 
     embed.set_thumbnail(url='https://api.scryfall.com/cards/named?exact={card}&format=image&version=art_crop'.format(
-        card=output['data'][0]['main_cards'][0].replace(' ', '%20')))
+        card=data['sample'][0]['main_card'].replace(' ', '%20')))
     for x in arr:
         embed.add_field(name=x['name'], value=x['value'], inline=False)
     await ctx.send(embeds=[embed])
 
 @drc.subcommand('matchups')
-@slash_option('args', 'No description given', OptionTypes.STRING)
-async def matchups(ctx: MtgContext, args: str) -> None:
-    """Matchup calculation using Dreadrise. Accepts two queries separated by exclamation mark !."""
-    q_list = args.split('!')
-    q1 = q_list[0]
-    q2 = q_list[1] if len(q_list) >= 2 else ''
-    count, output, error = await fetcher.dreadrise_search_matchups(q1, q2)
-    if error:
-        await ctx.send(f'Search error: `{error}`')
+@slash_option('q1', 'The query for the first player', OptionTypes.STRING, required=True)
+@slash_option('q2', 'The query for the second player', OptionTypes.STRING, required=True)
+async def matchups(ctx: MtgContext, q1: str, q2: Optional[str]) -> None:
+    """Matchup calculation using Dreadrise."""
+    await ctx.defer()
+    q2 = q2 or ''
+    data = await fetcher.dreadrise_search_matchups(q1, q2, 1)
+    if not data['success']:
+        await ctx.send('Search error: `{error}`'.format(error=data['reason']))
         return
-    if count == 0:
+    if data['matches'] == 0:
         await ctx.post_nothing()
         return
 
-    ans = '{length} matches found. Winrate: {wr}%\n{domain}/minimize/{url}'.format(
-        domain=link_domain, length=count, wr=output['winrate'], url=output['compress'])
+    ans = '{length} matches found. Winrate: {wr}%\n{domain}{path}?q1={q1}&q2={q2}{pd}'.format(
+        domain=link_domain,
+        length=data['matches'],
+        wr=data['winrate'],
+        path='/deck-search/matchups',
+        pd='&dist=penny_dreadful',
+        q1=q1.replace(' ', '%20'),
+        q2=q2.replace(' ', '%20'),
```

Comment:
You want to `urllib.parse.quote` here rather than this simple replace. This defends you against (for example) input containing an ampersand ('&') char.

Reply:
good call